containers/src/Kokkos_DualView.hpp

```diff
@@ -655,10 +662,21 @@ class DualView : public ViewTraits<DataType, Properties...> {
   }
 
   template <class ExecSpace>
-  void sync_host(const ExecSpace& exec) {
-    sync_host_impl(exec);
+  void sync_host([[maybe_unused]] const ExecSpace& exec) {
+    if constexpr (impl_dualview_stores_single_view)
+      return;
+    else
+      sync_host_impl(exec);
+  }
+  void sync_host() {
+    if constexpr (impl_dualview_stores_single_view) {
+      Kokkos::fence(
+          "Kokkos::DualView: fence for sync_host with host-accessible memory "
+          "space");
+      return;
+    } else
+      sync_host_impl();
   }
-  void sync_host() { sync_host_impl(); }
```

Comment:
Why is this removed? (same with `sync_device()`)
If a synchronous `sync()` is provided along with an asynchronous `sync(ExecSpace)`, wouldn't it be consistent to also supply synchronous `sync_[host|device]()` along with asynchronous `sync_[host|devce](ExecSpace)`?

Reply:
Same here. Ignoring white spache changes, I only added some fences.